### PR TITLE
Automate build and publish gem

### PR DESCRIPTION
**Pull request by punitcse**
# What

Every time merge with master it builds and publish the ruby gem.

# Why 
So we dont have to..

@simplybusiness/application-tooling

### CODE REVIEW

File: .github/workflows/gem-publish.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      - name: Set up Ruby 2.6
+        uses: actions/setup-ruby@v1
+        with:
+          version: 2.6.x
+      - name: Publish to RubyGems
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
```
Comment (moltenice) [inline .github/workflows/gem-publish.yml:22]: This shouldn't print the key out to stdout or log it at all right?

Comment (punitcse) [reply .github/workflows/gem-publish.yml:22]: No it does not. I had the same doubt :)

**Review by moltenice — APPROVED** [review]
💎

**Comment by moltenice** [comment]
Actually I think we should make another GitHub Action that fails unless versions.rb has changed. To ensure every PR has updated version.rb.

**Comment by sleepyfox** [comment]
Good call

**Comment by punitcse** [comment]
> Actually I think we should make another GitHub Action that fails unless versions.rb has changed. To ensure every PR has updated version.rb.

Sure I've checked but could not find a straightforward solution. I will check that and create a new PR for that.

**Pull request merged**